opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -205,13 +237,11 @@ namespace Opm {
     template<typename TypeTag>
     void
     BlackoilWellModel<TypeTag>::
-    assembleWellEq(Simulator& ebosSimulator,
-                   const double dt,
-                   WellState& well_state,
-                   bool only_wells) const
+    assembleWellEq(const double dt,
+                   bool only_wells)
     {
-        for (int w = 0; w < number_of_wells_; ++w) {
-            well_container_[w]->assembleWellEq(ebosSimulator, dt, well_state, only_wells);
+        for (int w = 0; w < wells()->number_of_wells; ++w) {
```

Comment:
The previous efforts were made to only use `Wells` struct in the initialization, maybe also one place related to group control. The point is that when we want to remove the `Wells` struct, we can do that relatively easily. I think it was something people like to have. Maybe in this PR due to WellState, we need a few more places to use it, but please do not undo other related efforts.

Reply:
now I saw `Wells` struct was also used in `resetWellControlFromState`, it is something can be changed to use the `well_container_`.

Reply:
The point of this change is to reduce the amount of redundant -- and thus potentially inconsistent -- information. I'm not aware of anybody who likes `struct Wells`, but with this PR it becomes an completely internal implementation detail of `WellModel`, so the should not harm any removal efforts?

Reply:
To avoid inconsistency, I will suggest 
```
int numWells() const
{
    return wells()->number_of_wells;
}

int numPhases() const
{
    return wells()->number_of_phases;
}
```
We have store the WellControls for each individual well, it looks like these are the only information we need from the `Wells struct`.

Reply:
I added numWells() and numPhases() to minimize the usage of wells(). Removing all traces of wells() is for another PR and day.